src/cobra/flux_analysis/deletion.py

```diff
@@ -405,3 +401,60 @@ def double_gene_deletion(
         processes=processes,
         **kwargs
     )
+
+
+@pd.api.extensions.register_dataframe_accessor("knockout")
+class KnockoutAccessor:
+    """Access unique combinations of reactions in deletion results."""
+
+    def __init__(self, pandas_obj):
+        """Set up the accessor.
+
+        Parameters:
+        -----------
+        pandas_obj : pd.DataFrame or pd.Series
+            A result from one of the deletion methods.
+        """
+        self._validate(pandas_obj)
+        self._result = pandas_obj
+
+    @staticmethod
+    def _validate(obj):
+        # verify it is a deletion results
+        if any(name not in obj.columns for name in ["ids", "growth", "status"]):
+            raise AttributeError("Must be DataFrame returned by a deletion method.")
+
+    def __getitem__(self, args):
+        """Return the deletion result for a particular set of knocked entities.
+
+        Parameters:
+        -----------
+        args : iterable of str, cobra.Reaction or cobra.Gene
+
+        Returns:
+        --------
+        pd.DataFrame
+            The deletion result where the chosen entities have been deleted.
+        """
+        if not any(isinstance(args, t) for t in [tuple, list]):
+            args = [args]
+
+        if any(isinstance(args[0], t) for t in [Reaction, Gene, str]):
+            try:
+                args = [{obj.id} for obj in args]
+            except AttributeError:
+                # are already strings
+                args = [{obj} for obj in args]
+        elif isinstance(args[0], set):
+            try:
+                args = [set(elem.id for elem in obj) for obj in args]
+            except AttributeError:
+                args = [set(obj) for obj in args]
+        else:
+            raise ValueError(
+                "Allowed indices are single Reactions or Genes, "
+                "lists of Reactions of Genes, or lists of sets "
+                "of Reactions or Genes."
+            )
+        found = [x in args for x in self._result.ids]
```

Comment:
`args` being a list of sets here, I think this will perform rather poorly. Could you not use `self._result.ids.isin(args)` or does that fail because sets are not hashable?

Reply:
I see it doesn't work because they are unhashable.